### PR TITLE
Add raisr into cartwheel

### PR DESCRIPTION
**Pull request by wenbinc-Bin**
Opencl version of raisr is released, so move raisr patches to carthweel. https://github.com/OpenVisualCloud/Video-Super-Resolution-Library

### CODE REVIEW

File: patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch [hunk 1]
```diff
@@ -0,0 +1,326 @@
+From 5399c850b6e16557cd9cfe71978b160332eaa80f Mon Sep 17 00:00:00 2001
+From: Wenbin Chen <wenbin.chen@intel.com>
+Date: Thu, 6 Apr 2023 13:48:23 +0800
+Subject: [PATCH 2/2] libavfilter/raisr_opencl: Add raisr_opencl filter
+
+Add raisr_opencl filter. It uses the same algorithm as raisr filter but
+it uses cl_mem as input and output, which helps to build high speed
+pipeline.
+For details please refer to https://github.com/intel-innersource/applications.media.ai.raisr-native-library
```
Comment (xhaihao) [inline patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch:9]: Please fix this

File: patches/0132-ffmpeg-raisr-filter.patch [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+ extern const AVFilter ff_vf_readeia608;
+ extern const AVFilter ff_vf_readvitc;
+diff --git a/libavfilter/vf_raisr.c b/libavfilter/vf_raisr.c
+new file mode 100644
+index 0000000000..854703e790
+--- /dev/null
++++ b/libavfilter/vf_raisr.c
+@@ -0,0 +1,344 @@
++/*
++ * Intel Library for Video Super Resolution ffmpeg plugin
```
Comment (xhaihao) [inline patches/0132-ffmpeg-raisr-filter.patch:98]: Is this needed ?

File: patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch (hunk 1 above)
Comment (wenbinc-Bin) [reply patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch]: Good catch! Done.

File: patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
++        vdt_out[p].step = output->linesize[p];
++        vdt_out[p].bitShift = desc->comp[p].shift;
++    }
++
++    if (!ctx->initialised) {
++        err = RNLHandler_SetRes(&vdt_in[0], &vdt_in[1], &vdt_in[2],
++                                &vdt_out[0], &vdt_out[1], &vdt_out[2]);
++        if (err != RNLErrorNone) {
++            av_log(ctx, AV_LOG_ERROR, "RNLHandler_SetRes error\n");
++            return AVERROR(ENOMEM);
```
Comment (xhaihao) [inline patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch:185]: both input and output should be freed,

File: patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
++        }
++        ctx->initialised = 1;
++    }
++
++    err = RNLHandler_Process(&vdt_in[0], &vdt_in[1], &vdt_in[2],
++                             &vdt_out[0], &vdt_out[1], &vdt_out[2],
++                             ctx->blending);
++    if (err != RNLErrorNone) {
++        av_log(ctx, AV_LOG_ERROR, "RNLHandler_Process error\n");
++        return AVERROR(ENOMEM);
```
Comment (xhaihao) [inline patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch:195]: same as above.

File: patches/0132-ffmpeg-raisr-filter.patch [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
++        &vdt_in[2],
++        &vdt_out[0],
++        &vdt_out[1],
++        &vdt_out[2],
++        raisr->blending);
++
++    if (ret != RNLErrorNone)
++    {
++        av_log(ctx, AV_LOG_INFO, "RNLHandler_Process error\n");
++        return AVERROR(ENOMEM);
```
Comment (xhaihao) [inline patches/0132-ffmpeg-raisr-filter.patch:390]: in and out are not freed in the error path.

File: patches/0132-ffmpeg-raisr-filter.patch (hunk 2 above)
Comment (wenbinc-Bin) [reply patches/0132-ffmpeg-raisr-filter.patch:98]: We need it. Maybe some server user want to use cpu version raisr.

File: patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch (hunk 3 above)
Comment (wenbinc-Bin) [reply patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch]: Done

File: patches/0132-ffmpeg-raisr-filter.patch (hunk 5 above)
Comment (wenbinc-Bin) [reply patches/0132-ffmpeg-raisr-filter.patch:390]: Done

File: patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch (hunk 4 above)
Comment (wenbinc-Bin) [reply patches/0133-libavfilter-raisr_opencl-Add-raisr_opencl-filter.patch]: done

**Review by xhaihao — APPROVED** [review]
LGTM

**Pull request merged**